### PR TITLE
Fix a few typos

### PR DESCRIPTION
**Pull request by jdkato**
These were found by [Vale](https://github.com/errata-ai/vale), which was mentioned as a candidate for https://github.com/Khan/engblog/issues/51. If you're interested, I could submit another PR that adds Vale to your CI build.

The PR would include the following:

1. A `.vale.ini` file configured to lint Markdown and reStructuredText documents, excluding front matter and math (both `$$ … $$` and `$ … $`);
2. a custom vocabulary file that you can add to as you encounter words that Vale doesn't recognize; and
3. a simple `engblog` style to catch common capitalization mistakes like "Javascript" instead of "JavaScript."

For what it's worth, [Linode](https://github.com/linode/docs) and [Write the Docs](https://github.com/writethedocs/www) have been using Vale in a similar capacity for a few months now.

### CODE REVIEW

File: src/posts/snippet-server.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ that.  Contributors welcome!
 
 We believe that snippets can be a useful tool for a small to
 medium-sized team, and while there are several snippet server
-implementations out there, this one's ease of use and low low price
```
Comment (csilvers) [inline src/posts/snippet-server.md]: This one is intentional, can you revert the fix?

File: src/posts/memcached-fms.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ for item in content_items:
...
+This was a pretty effective method &ndash; on our first round of testing, half of the hundred URLs we hit were at least 50% slower at the 50th percentile, and `khanacademy.org/math` was over 7 times slower!  But after several rounds of fixes and new tests, we were able to get the new version faster on average and on most specific URLs.  Unfortunately, this didn't cover everything &ndash; some real user traffic is harder to simulate, and because caching is less effective on test versions with little traffic, there was a lot of noise in our small sample, and so small changes in performance were hard to spot, even if they affected very popular routes where that small difference would add up.
```
Comment (csilvers) [inline src/posts/memcached-fms.md]: What's the difference here?  -- github isn't highlighting anything.

Comment (csilvers) [reply src/posts/memcached-fms.md]: Ah, I see: s/very very/very/.  As below, the doubling is intentional here and this should be reverted.

**Review by csilvers — CHANGES_REQUESTED** [review]
Overall these fixes look good!  How many false positives did vale have that you ignored?  In general, I'm sure automated grammar checkers are better than they once were but I wouldn't be surprised if they still had a lot of false hits.  It probably depends on the individual post-writer whether they'd find vale useful or not . I'm not familiar with the product; it is opt in?  How would we use it?

**Comment by jdkato** [comment]
I addressed your change requests in 23dbfccc1f75b8b4c5aa8593722de7539a1a94a5.

As for Vale itself, it's really not a grammar checker&mdash;it's much more like traditional source code linters. In other words, outside of simple things like spelling and repetition, Vale focuses on stylistic errors (e.g., branding, heading styles, word choice, etc.) rather than grammatical or structural errors. 

Anyways, with the configuration I mentioned above, this was Vale's exact output: 

<details><summary>Vale output</summary><p>


```console
$ vale --glob='*.{md,rst}' src/posts/

src/posts/creating-query-components-with-apollo.md
 14:49  warning  'in' is repeated!  vale.Repetition 

src/posts/copy-pasting-more-than-just-text.md
 64:61  error  Use 'JavaScript' instead of     engblog.Terms 
               'Javascript'.                                 
 97:36  error  Use 'JavaScript' instead of     engblog.Terms 
               'Javascript'.                                 

src/posts/evolving-our-content-infrastructure.md
 398:56  warning  'had' is repeated!  vale.Repetition 

src/posts/fun-with-slopfields.md
 75:280  error  Did you really mean             engblog.Spelling 
                'efectively'?                                    

src/posts/lets-reduce.md
 21:83  error  Use 'JavaScript' instead of     engblog.Terms 
               'Javascript'.                                 
 211:1  error  Use 'JavaScript' instead of     engblog.Terms 
               'Javascript'.                                 

src/posts/memcached-fms.md
 41:637  warning  'very' is repeated!  vale.Repetition 

src/posts/new-employees-primer.md
 87:9  warning  'ship' is repeated!  vale.Repetition 

src/posts/new-oss-activity.md
 17:88   error    Did you really mean             engblog.Spelling 
                  'dependancy'?                                    
 17:150  warning  'which' is repeated!            vale.Repetition  
 30:416  error    Did you really mean             engblog.Spelling 
                  'dyanamic'?                                      
 30:425  error    Did you really mean             engblog.Spelling 
                  'prorogramming'?                                 

src/posts/python-refactor-3.md
 85:45  warning  'from' is repeated!  vale.Repetition 

src/posts/react-native-monorepo.md
 187:24  error  Did you really mean             engblog.Spelling 
                'suprised'?                                      

src/posts/receiving-feedback.md
 27:463  warning  'that' is repeated!  vale.Repetition 

src/posts/snippet-server.md
 23:1    error    Use 'JavaScript' instead of     engblog.Terms   
                  'javascript'.                                   
 159:55  warning  'low' is repeated!              vale.Repetition
```
</p></details>

<br>

With regards to how it can be used, the basic idea would be to have it run via Travis CI (and cause the build to fail on error-level alerts) whenever you make content changes to `src/posts`. It would have caught [most of these](https://github.com/Khan/engblog/pulls?utf8=%E2%9C%93&q=typo) (outside of "an" and "its").

**Comment by csilvers** [comment]
Well, I'm not sure exactly what "word choice" it would be unhappy with, but I definitely have no problem with the fixes it proposed here.  Thanks for making them!

Let's continue the discussion about whether to install vale on that issue.

**Pull request merged**